runtime/apps/spdm/spdm-lib/src/protocol/algorithms.rs

```diff
@@ -3,8 +3,84 @@
 use crate::error::{SpdmError, SpdmResult};
 use bitfield::bitfield;
 use zerocopy::{FromBytes, Immutable, IntoBytes};
-pub(crate) trait Prioritize<T> {
-    fn prioritize(self, peer: Self, priority_table: Option<&[T]>) -> Self;
+pub(crate) trait Prioritize<T>
+where
+    Self: Sized,
+    T: Copy + Into<Self>,
+{
+    fn prioritize(&self, peer: &Self, priority_table: Option<&[T]>) -> Self;
+}
+
+impl<T> Prioritize<T> for u8
+where
+    T: Copy + Into<u8>,
+{
+    fn prioritize(&self, peer: &Self, priority_table: Option<&[T]>) -> Self {
+        let common = self & peer;
+        if let Some(priority_table) = priority_table {
+            for &priority in priority_table {
+                let priority_spec: u8 = priority.into();
+                if common & priority_spec != 0 {
+                    return common & (!common + 1);
```

Comment:
I think this (and the ones below) should be?

```suggestion
                    return 1 << (common & priority_spec).trailing_zeros();
```